gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -133,8 +133,22 @@ private <T> T withRevWalkUnchecked(CheckedFunction1<RevWalk, T> fun) {
     }
   }
 
-  private boolean isBranchPresent(String branchFullName) {
-    return Try.of(() -> jgitRepoForMainGitDir.resolve(branchFullName)).getOrNull() != null;
+  private boolean isBranchPresent(String branchFullName, String currentlyCheckedBranchFullName) {
+    boolean branchExists = Try.of(() -> jgitRepoForMainGitDir.resolve(currentlyCheckedBranchFullName)).getOrNull() != null;
+    if (branchFullName.equals(currentlyCheckedBranchFullName)) {
```

Comment:
Whatever the implementation ends up like, pls document why this workaround was needed in the first place as it's non-trivial as f**k